### PR TITLE
Add option for ignoring certain monitor slugs

### PR DESCRIPTION
**Pull request by adinauer**
## :scroll: Description

Add the following options:
- `enableAutomaticCheckIns` which enables automatic Check-ins, e.g. Quartz; defaults to `false` meaning it's disabled
- `ignoredCheckIns` which can be used to set a list of monitor slugs and regular expressions matching on the Check-ins monitor slug. These will then be dropped in `SentryClient`. 

## :bulb: Motivation and Context

Opt-in to automatic check-ins and a way to filter check-ins.

## :green_heart: How did you test it?
Manually + unit tests

## :pencil: Checklist

- [ ] I reviewed the submitted code.
- [ ] I added tests to verify the changes.
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [ ] I updated the docs if needed.
- [ ] Review from the native team if needed.
- [ ] No breaking change or entry added to the changelog.
- [ ] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

File: sentry-spring-boot-jakarta/src/test/kotlin/io/sentry/spring/boot/jakarta/SentryAutoConfigurationTest.kt [hunk 1]
```diff
@@ -724,6 +732,41 @@ class SentryAutoConfigurationTest {
...
+    fun `when auto checkins is enabled, does not create quartz config if spring-quartz lib missing`() {
+        contextRunner.withPropertyValues("sentry.dsn=http://key@localhost/proj", "sentry.enable-automatic-checkins=true")
+            .withClassLoader(FilteredClassLoader(SchedulerFactoryBean::class.java))
+            .run {
+                assertThat(it).doesNotHaveBean(SchedulerFactoryBeanCustomizer::class.java)
+            }
+    }
+
+    @Test
+    fun `when auto checkins is enabled, does not create quartz config if sentry-quartz lib missing`() {
```
Comment (lbloder) [inline sentry-spring-boot-jakarta/src/test/kotlin/io/sentry/spring/boot/jakarta/SentryAutoConfigurationTest.kt:762]: Maybe add a `disabled` test?

File: sentry-spring-boot/src/test/kotlin/io/sentry/spring/boot/SentryAutoConfigurationTest.kt [hunk 2]
```diff
@@ -724,6 +732,41 @@ class SentryAutoConfigurationTest {
...
+    fun `when auto checkins is enabled, does not create quartz config if spring-quartz lib missing`() {
+        contextRunner.withPropertyValues("sentry.dsn=http://key@localhost/proj", "sentry.enable-automatic-checkins=true")
+            .withClassLoader(FilteredClassLoader(SchedulerFactoryBean::class.java))
+            .run {
+                assertThat(it).doesNotHaveBean(SchedulerFactoryBeanCustomizer::class.java)
+            }
+    }
+
+    @Test
+    fun `when auto checkins is enabled, does not create quartz config if sentry-quartz lib missing`() {
```
Comment (lbloder) [inline sentry-spring-boot/src/test/kotlin/io/sentry/spring/boot/SentryAutoConfigurationTest.kt:762]: Maybe add a `disabled` test?

File: sentry/src/main/java/io/sentry/SentryOptions.java [hunk 3]
```diff
@@ -2159,6 +2174,34 @@ public void setSendModules(boolean sendModules) {
...
+    this.enableAutomaticCheckIns = enableAutomaticCheckIns;
+  }
+
+  public void setIgnoredCheckIns(final @Nullable List<String> ignoredCheckIns) {
+    if (ignoredCheckIns == null) {
+      this.ignoredCheckIns = null;
+    } else {
+      @NotNull final List<String> filteredIgnoredCheckIns = new ArrayList<>();
+      for (String slug : ignoredCheckIns) {
+        if (!slug.isEmpty()) {
```
Comment (lbloder) [inline sentry/src/main/java/io/sentry/SentryOptions.java:2192]: should we also check for `blank` here?

File: sentry/src/main/java/io/sentry/SentryOptions.java [hunk 4]
```diff
@@ -2407,6 +2450,14 @@ public void merge(final @NotNull ExternalOptions options) {
     if (options.isSendModules() != null) {
       setSendModules(options.isSendModules());
     }
+
+    if (options.isEnableAutomaticCheckIns() != null) {
+      setEnableAutomaticCheckIns(options.isEnableAutomaticCheckIns());
+    }
+    if (options.getIgnoredCheckIns() != null) {
+      final List<String> ignoredCheckIns = new ArrayList<>(options.getIgnoredCheckIns());
+      setTracePropagationTargets(ignoredCheckIns);
```
Comment (lbloder) [inline sentry/src/main/java/io/sentry/SentryOptions.java:2459]: Probably CopyPaste error. Ignored check-ins shouldn't be added to `tracePropagationTargets`, instead `setIgnoredCheckIns` should be called

**Review by lbloder — COMMENTED** [review]
Some comments, LGTM otherwise 👍

File: sentry/src/main/java/io/sentry/SentryOptions.java (hunk 3 above)
Comment (adinauer) [reply sentry/src/main/java/io/sentry/SentryOptions.java:2192]: I just copied what we do for tracing origins so I'd expect this to be sufficient. Can always improve later.

File: sentry/src/main/java/io/sentry/SentryOptions.java (hunk 4 above)
Comment (adinauer) [reply sentry/src/main/java/io/sentry/SentryOptions.java:2459]: good catch! Seems I missed adding tests to `SentryOptionsTest` it seems. adding those as they should also catch this

**Review by lbloder — APPROVED** [review]
LGTM

**Pull request merged**